### PR TITLE
Enable cache for NXP S32Z/E SoCs and boards

### PR DESCRIPTION
**Pull request by duongvunam**
This PR enables support for caches on Arm Cortex-R52 cores and enables its usage on NXP S32 Z/E SoCs, specifically for S32Z27x-DC(2) board.

Add I/D caches capability to Cortex-R52 and enable them at NXP S32Z/E Socs. Support nocache for Cortex-R52 and enable nocache in each driver using it. Refactor and configure  MPU to configure static region (nocache is on this region ). 

Update alignment for Cortex-R52 since Cortex-R52 doesn't require power of 2 alignments.

### CODE REVIEW

File: arch/arm/core/aarch32/mpu/arm_mpu.c [hunk 1]
```diff
@@ -350,11 +350,17 @@ int z_arm_mpu_init(void)
 	/* Clean and invalidate data cache if it is enabled and
 	 * that was not already done at boot
 	 */
+#if (CONFIG_CPU_AARCH32_CORTEX_R)
```
Comment (carlocaione) [inline arch/arm/core/aarch32/mpu/arm_mpu.c:353]: `if defined(`

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: arch/arm/core/aarch32/mpu/arm_mpu.c (hunk 1 above)
Comment (duongvunam) [reply arch/arm/core/aarch32/mpu/arm_mpu.c]: I updated

**Review by carlocaione — DISMISSED** [review]
LGTM for the cache part. Cannot vouch for the MPU changes.

**Comment by microbuilder** [comment]
@povergoing Mind having a look since you're more familiar with this core?

**Review by manuargue — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]
LGTM for the MPU part. But just a kind of reminder, disabling/enabling MPU every time in context switch might cause a performance issue.

**Review by microbuilder — DISMISSED** [review]

**Comment by manuargue** [comment]
> disabling/enabling MPU every time in context switch might cause a performance issue.

Indeed. This needs to be addressed in a separate PR as an improvement.

**Review by danieldegrasse — DISMISSED** [review]

**Review by manuargue — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Comment by povergoing** [comment]
The CI fails but seems not caused by this  PR

**Comment by duongvunam** [comment]
updated your branch to remove a merge commit

**Review by manuargue — DISMISSED** [review]

**Comment by manuargue** [comment]
> The CI fails but seems not caused by this PR

Fixed in. Pls rebase @duongvunam to fix CI failures on this pr

File: arch/arm/core/aarch32/mpu/arm_mpu.c (hunk 1 above)
Comment (povergoing) [reply arch/arm/core/aarch32/mpu/arm_mpu.c:353]: @duongvunam, seems not fixed with `if defined(`

Comment (duongvunam) [reply arch/arm/core/aarch32/mpu/arm_mpu.c]: fixed with if defined(

**Review by povergoing — APPROVED** [review]

**Review by manuargue — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**